kore/src/Kore/Strategies/Goal.hs

```diff
@@ -879,11 +894,9 @@ derivePar rules goal = errorBracket $ do
     configuration :: Pattern Variable
     configuration = getConfiguration goal
 
-    errorBracket action =
-        onException action
-            (formatExceptionInfo
-                ("configuration=" <> unparseToText configuration)
-            )
+    enrichEventualException action =
+        catch action $
```

Comment:
If you flip the arguments of `GoalException` and remove the type parameter, this function is simply

```suggestion
        handle (throw . GoalException configuration)
```